### PR TITLE
unbraced variable must not apply advanced substitutions

### PR DESCRIPTION
**Pull request by ndeloof**
close https://github.com/docker/compose/issues/8601

aligns with https://github.com/docker/compose/blob/2011bc3cea1c78eccd0fa75d01a6b638d6bfb7c2/compose/config/interpolation.py#L96-L98

### CODE REVIEW

File: template/template.go [hunk 1]
```diff
@@ -25,11 +25,12 @@ import (
 )
 
 var delimiter = "\\$"
-var substitution = "[_a-z][_a-z0-9]*(?::?[-?][^}]*)?"
+var id = "[_a-z][_a-z0-9]*"
+var bid = "[_a-z][_a-z0-9]*(?::?[-?][^}]*)?"
```
Comment (mat007) [inline template/template.go:29]: What does `bid` mean?

**Review by mat007 — APPROVED** [review]
Makes sense!

File: template/template.go (hunk 1 above)
Comment (ulyssessouza) [reply template/template.go:29]: Maybe naming it `substitutionNamed` and `substitutionBraced`

Comment (ndeloof) [reply template/template.go:29]: "braced ID" I guess?
My intent was to align with https://github.com/docker/compose/blob/2011bc3cea1c78eccd0fa75d01a6b638d6bfb7c2/compose/config/interpolation.py#L96-L98

**Review by ulyssessouza — APPROVED** [review]
nit: regexp naming.

**Pull request merged**